cmd/xl-storage-format-v2.go

```diff
@@ -1788,7 +1788,7 @@ func (x xlMetaV2) ToFileInfo(volume, path, versionID string, inclFreeVers, allPa
 			}
 		}
 		if found {
-			continue
+			break
```

Comment:
I think we need to continue, for now. It looks like we are counting the number of free versions in this xl.meta so we can report correctly the number of versions in line fi.NumVersions (not sure why NumVersions is used, mostly for ILM max non current versions count)

Reply:
Okay then this is a problem @krisis we need to see why this is the case.

Reply:
Not really. It is just checking the flag for each entry. Even with 100K versions that is not really going to be slow.

Reply:
Okay then, let's leave this for now.